syntax="proto2";
import "pds_option.proto";
package baidu.infinite.map_model;
option cc_generic_services = true;
message Tensor {
    required string name = 1;
    repeated uint32 shape = 2;
    required bytes features = 3;
};

message SparseTensor {
    required string name = 1;
    repeated uint32 keys = 2;
    repeated uint32 shape = 3;
    required bytes features = 4;
};

message DenseInstance {
    repeated Tensor tensors = 1;
};

message SparseInstance {
    repeated SparseTensor tensors = 1;
};

message DenseRequest {
    repeated DenseInstance instances = 1;
};

message Request {
    repeated SparseInstance instances = 1;
};
message DensePrediction {
    repeated float categories = 1;
};
message Response {
    repeated DensePrediction predictions = 1;
};
service MapCnnService {
    rpc inference(Request) returns (Response);
    rpc debug(Request) returns (Response);
    option (pds.options).generate_stub = true;
};

service MapDenseCnnService {
    rpc inference(DenseRequest) returns (Response);
    rpc debug(DenseRequest) returns (Response);
    option (pds.options).generate_stub = true;
};
